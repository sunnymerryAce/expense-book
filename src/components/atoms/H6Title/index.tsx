import React from "react";
import Typography from "@material-ui/core/Typography";
import useStyles from "./style";

interface H6TitleProps {
  text: string;
}

const H6Title: React.FC<H6TitleProps> = (props) => {
  const classes = useStyles();
  return (
    <Typography variant="h6" className={`H6Title ${classes.root}`}>
      {props.text}
    </Typography>
  );
};

export default H6Title;
