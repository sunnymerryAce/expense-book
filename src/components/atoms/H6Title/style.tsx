import { makeStyles, createStyles } from "@material-ui/core/styles";

const useStyles = makeStyles(() =>
  createStyles({
    root: {
      textAlign: "center",
      padding: "20px 0 0",
    },
  })
);

export default useStyles;
