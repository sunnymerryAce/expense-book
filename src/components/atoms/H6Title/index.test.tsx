import React from "react";
import H6Title from "./";
import { shallow } from "enzyme";

describe("<H6Title />", () => {
  let container: any = null;
  const props = {
    text: "title",
  };

  beforeEach(() => {
    container = shallow(<H6Title {...props} />);
  });

  afterEach(() => {
    container.unmount();
    container = null;
  });

  test("should match the snapshot", () => {
    expect(container.html()).toMatchSnapshot();
  });

  test("should show proper text", () => {
    expect(container.find(".H6Title").text()).toBe(props.text);
  });
});
