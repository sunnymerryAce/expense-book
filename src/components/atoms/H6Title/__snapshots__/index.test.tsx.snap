// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<H6Title /> should match the snapshot 1`] = `"<h6 class=\\"MuiTypography-root H6Title makeStyles-root-1 MuiTypography-h6\\">title</h6>"`;
